agent/src/beerocks/slave/link_metrics/ieee802_3_link_metrics_collector.cpp

```diff
@@ -0,0 +1,279 @@
+                result = true;
+            }
+            break;
+        case IFLA_STATS:
+            if (result) {
+                struct rtnl_link_stats *stats = (struct rtnl_link_stats *)RTA_DATA(attribute);
+
+                /**
+                 * Get interface speed into PHY rate.
+                 */
+                uint32_t phy_rate_mbps = UINT32_MAX;
+                beerocks::net::network_utils::linux_iface_get_speed(local_interface_name,
+                                                                    phy_rate_mbps);
+
+                link_metrics.transmitter.packet_errors       = stats->tx_errors;
+                link_metrics.transmitter.transmitted_packets = stats->tx_packets;
+                /**
+                 * Note: The MAC throughput capacity is a function of the physical data rate and
+                 * of the MAC overhead. We could somehow compute such overhead or, for simplicity,
+                 * set the MAC throughput as a percentage of the physical data rate.
+                 * For Ethernet, we can estimate the overhead: 7 bytes preamble, 1 byte SFD, 14
+                 * bytes header, 4 bytes CRC and 12 bytes of interpacket gap on a 1500 byte
+                 * payload. So 1500/1538.
+                 * (see https://en.wikipedia.org/wiki/Ethernet_frame)
+                 */
+                const uint32_t layer2_payload_size = 1500;
+                const uint32_t layer1_total_size   = 1538;
+                link_metrics.transmitter.mac_throughput_capacity_mbps =
+                    phy_rate_mbps * ((float)layer2_payload_size / (float)layer1_total_size);
+                // Note: For simplicity, link availability is set to "100% of the time"
+                link_metrics.transmitter.link_availability = 100;
+                link_metrics.transmitter.phy_rate_mbps     = phy_rate_mbps;
+
+                link_metrics.receiver.packet_errors    = stats->rx_errors;
+                link_metrics.receiver.packets_received = stats->rx_packets;
+                link_metrics.receiver.rssi             = UINT8_MAX;
+            }
+            break;
+        }
+    }
+
+    return result;
+}
+
+/**
+ * @brief Gets link metrics of an Ethernet network interface.
+ *
+ * Gets link metrics for given Ethernet network interface by sending a RTM_GETLINK Netlink request
+ * through the specified Netlink socket and parsing received response.
+ *
+ * @param[in] fd File descriptor of a connected Netlink socket.
+ * @param[in] local_interface_name Name of the Ethernet network interface.
+ * @param[in, out] link_metris Link metrics structure with read values.
+ *
+ * @return True on success and false otherwise.
+ */
+static bool get_link_metrics(int fd, const std::string &local_interface_name,
+                             sLinkMetrics &link_metrics)
+{
+    bool result = false;
+
+    struct sockaddr_nl kernel; /* the remote (kernel space) side of the communication */
```

Comment:
I think `socket` or `sock` is a more appropriate name here.
```suggestion
    struct sockaddr_nl sock; /* the remote (kernel space) side of the communication */
```
(nitpick)

Reply:
Done